dev_tool/src/jumpbox.ts

```diff
@@ -0,0 +1,323 @@
+    )
+    if (!securityGroupID || securityGroupID.GroupId === undefined) {
+        return new Error('No security groups on the instance to update')
+    }
+
+    // get the right rule, check if my IP is in it
+    const securityGroup = await describeSecurityGroup(securityGroupID.GroupId)
+    if (securityGroup instanceof Error) {
+        return securityGroup
+    }
+
+    const port22Rules = securityGroup.IpPermissions?.find(
+        (perm) => perm.FromPort === 22
+    )
+    if (!port22Rules) {
+        return new Error(
+            'Security Group does not have port 22 on it, this will probably need to be fixed by hand, contact @mcrd'
+        )
+    }
+
+    const myRule = port22Rules.IpRanges?.find((range) =>
+        range.CidrIp?.startsWith(myIPAddress)
+    )
+    if (myRule) {
+        // if our IP is already whitelisted, we're good.
+        console.info('Already Whitelisted')
+        return undefined
+    }
+
+    // in this case, now we need to add our IP address.
+    console.info('Whitelisting', myIPAddress)
+
+    const result = await addSSHWhitelistRuleToGroup(
+        securityGroupID.GroupId,
+        myIPAddress
+    )
+    if (result instanceof Error) {
+        return result
+    }
+
+    return undefined
+}
+
+async function cloneDBLocally(envName: string, stopAfter = true) {
+    const check = await checkAWSAccess(envName)
+    if (check instanceof Error) {
+        process.exit(1)
+    }
+
+    // Figure out if Jumpbox is running
+    const instance = await ensureJumpboxIsRunning()
+    if (instance instanceof Error) {
+        console.error('Error setting IP whitelist', instance)
+        process.exit(1)
+    }
+
+    const whitelist = await ensureWhitelistIP(instance)
+    if (whitelist instanceof Error) {
+        console.error('Error setting IP whitelist', whitelist)
+        process.exit(1)
+    }
+
+    const jumpboxInstance = instance
+
+    const jumpboxIP = jumpboxInstance.PublicIpAddress
+    const jumpboxInstanceID = jumpboxInstance.InstanceId
+
+    if (!jumpboxIP || !jumpboxInstanceID) {
+        console.error(
+            'EC2 didnt return required information',
+            jumpboxIP,
+            jumpboxInstanceID
+        )
+        process.exit(1)
+    }
+
+    // Get the secrets for the DB.
+    const dbSecrets = await getSecretsForRDS(stageForEnv(envName))
+    if (dbSecrets instanceof Error) {
+        console.error('error fetching secrets', dbSecrets)
+        process.exit(1)
+    }
+
+    try {
+        // start an SSH connection
+        console.info('Connecting to ', jumpboxIP)
+        const ssh = new NodeSSH()
+
+        // If the jumpbox just started we often have the connection refused for ~10 seconds until it's really up.
+        const connectionResult = await retry(async () => {
+            try {
+                await ssh.connect({
+                    host: jumpboxIP,
+                    username: 'ubuntu',
+                    privateKeyPath: '/Users/macrae/.ssh/wml_jumpbox',
```

Comment:
Hmm, we're going to need to use each individual's private key here. I looked at the lib and it seems to require a path. There's [another lib](https://github.com/mcavage/node-ssh-agent) that can read from ssh-agent, but we should probably think about what to do here.